### ISSUE TITLE
Annotation Tags: 101st tag in alphabetical order is not displayed in drop-down menu any more

### ISSUE THREAD

**Issue by petrovskonek** [grafana/grafana#68211]
### 
# What went wrong?

A limited number of 100 tags is displayed in the tags drop-down menu under “Add Annotation”. It would be useful, if every tag that has been created by adding annotations is shown in the drop-down menu. We use the „Annotations & Alerts (Built-in)“. Unfortunately, only the first 100 tags in alphabetical order of approximately 140 tags that we have created so far can be chosen in the drop-down menu, if one types in the first letters of the given tag. The last 40 entries must therefore be typed in manually without that the given tag is suggested and can be chosen from the drop-down menu.

**What happened**:

  - As soon as we had more than 100 tag entries, the last entries in alphabetical order were not displayed in the drop-down menu any more. 

**What did you expect to happen**:

  - I expected the 101st entry to be displayed in the drop-down menu as the 100th entry. 

### 
# How do we reproduce it?

If you want to check it yourself, in the demo dashboard
https://play.grafana.org/d/000000012/grafana-play-home?orgId=1  
you can see that the last entry drops out of the list, if you add a tag that is listed above and is not suggested any more even if you type in the first letters of the “lost” tag. 
To be more precise: 
If I scroll down the drop-down menu of the tags in the given Dashboard, I get as last tag “#tocheck” (screenshot on the left) and if I add as next tag, the tag “#to”, because that comes just before it in the alphabetical order and save this tag (screenshot on the right)

![image](https://github.com/grafana/grafana/assets/132680322/53ff414c-3bb9-465f-8ce1-b21ae2c5b0e6)

for the next annotation, the last tag in the drop-down menu that is displayed is “#to” (screenshot on the left). Even if I type in the first letters “#to”, “#tocheck” is not suggested any more (screenshot on the right). 

If you cannot reproduce the bug, please try the following: 
Since the 100th comment tag appears in alphabetical order (first character of the string) but the 101st comment tag does not, and this changes when a user adds a new comment tag, one better tries “zzzz”.
I added the comment tag “zzzz” but it never appears in the search when typing “z”.

![image](https://github.com/grafana/grafana/assets/132680322/4ef8a31b-f668-4cdd-87f6-63d224c64bdd)

In this case the tags are short, but in our case, we have long tags that we need to type in which is unfortunately a bit annoying and time-consuming. Thus, it would be nice, if more tags than the first 100 in alphabetical order were suggested when typing in the first letters of a tag.

### 
# What Grafana version are you using?

I am using Grafana 8.5.5, but it is the same with 9.3.6 (Demo Dashboards)

### 
### Is the bug inside a Dashboard Panel?

Copy the panel's ["get-help" data](https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/) here

### Grafana Platform?

Grafana: 10.2.3

### Is this a Regression?

No

**Comment by usmangt** [comment]
Hi @petrovskonek,

Thanks for opening this issue.

I am also linking the community post [thread](https://community.grafana.com/t/show-more-than-100-tags/81271) here for reference.

For the Engineering team.

I was able to reproduce it (but it could be tricky). I feel there is definitely some inconsistency when the UI shows a list of Annotation tags (which the user also explained).

**Comment by JoaoSilvaGrafana** [comment]
Thanks for raising this @petrovskonek , I was able to reproduce it.
It seems like the endpoint has a limit of 100 tags and we do the filtering in the frontend. It might be possible to use an `AsyncSelect` instead to query the endpoint if it allows for filtering.

But this is more part of "Panel annotations" than "Annotations list panel", so I am moving it to @grafana/dashboards-squad , let us know if we can help with anything Select-wise.

**Comment by evictorero** [comment]
The backend team for dashboards would be @grafana/backend-platform.
The 100 limit seems to be a safe default value but the backend already supports a query param `limit`.
Implementing an async search is the way to go but If they don't have the bandwidth to implement this search we could add a query param with a bigger limit value like 200 or 300 from the front end to support these edge cases.

**Comment by tolzhabayev** [comment]
@evictorero I am not quite sure what the question towards backend platform is?
Implementing `AsyncSelect` in the dashboards frontend is something that the dashboards squad should take a look at. I don't think we are missing anything in the API that would be necessary to do that.

**Comment by evictorero** [comment]
That's right, the existing API should be enough. Sorry for the confusion.
